File: kq-projection-dotnet/kq-projection/Common/Models/Events/GameStartedEventData.cs
namespace KQ_Projection.Common.Models.Events;

using KQ_Projection.Common.Models.Jigsaw;
using System.Text.Json.Serialization;

public class GameStartedEventData : EventData
{
    [JsonPropertyName("puzzleImageId")]
    public int puzzleImageId { get; set; }

    [JsonPropertyName("horizontalSize")]
    public int horizontalSize { get; set; }

    [JsonPropertyName("verticalSize")]
    public int verticalSize { get; set; }

    [JsonPropertyName("puzzlePieces")]
    public List<JigsawPuzzlePiece> puzzlePieces { get; set; }
}

File: java-apps/kq-common/src/main/java/com/kafkaquest/kq/common/util/models/events/eventdata/GameStartedEventData.java
package com.kafkaquest.kq.common.util.models.events.eventdata;

import com.kafkaquest.kq.common.util.models.events.other.JigsawPuzzlePiece;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@EqualsAndHashCode(callSuper = false)
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class GameStartedEventData extends EventData {

    private int puzzleImageId;

    private int horizontalSize;

    private int verticalSize;

    private List<JigsawPuzzlePiece> puzzlePieces;

}
